### ISSUE TITLE
hw_CANSocket exception when sending empty/malformed CAN frames

### ISSUE THREAD

**Issue by DePierre** [eik00d/CANToolz#83]
Hi there,

Nothing urgent there, only a first-time user experience I wanted to share with you.

When sending empty or invalid CAN frame, the web interface doesn't show any error. The exception is fully shown in the console (which is good IMO) but the web interface should at least display a generic error message.

Sending an empty frame will raise an exception in `cantoolz`:
```
Traceback (most recent call last):
  File "cantoolz.py", line 110, in do_POST
    text = self.can_engine.call_module(self.can_engine.find_module(str(path_parts[3])), str(paramz))
  File "/home/pop/can/CANToolz/cantoolz/engine.py", line 97, in call_module
    ret = self._enabledList[x][1].raw_write(params)
  File "/home/pop/can/CANToolz/cantoolz/module.py", line 60, in raw_write
    ret = cmd[3](cmd[5])
TypeError: dev_write() missing 1 required positional argument: 'data'
```

Sending a malformed frame like `aa` will raise:
```
Traceback (most recent call last):
  File "cantoolz.py", line 110, in do_POST
    text = self.can_engine.call_module(self.can_engine.find_module(str(path_parts[3])), str(paramz))
  File "/home/pop/can/CANToolz/cantoolz/engine.py", line 97, in call_module
    ret = self._enabledList[x][1].raw_write(params)
  File "/home/pop/can/CANToolz/cantoolz/module.py", line 58, in raw_write
    ret = cmd[3](cmd[5], lost_cmd)
  File "/home/pop/can/CANToolz/cantoolz/modules/hw_CANSocket.py", line 39, in dev_write
    idf, dataf = data.strip().split('#')
ValueError: not enough values to unpack (expected 2, got 1)
```

Sending a malformed frame like `a#a` will raise:
```
Traceback (most recent call last):
  File "cantoolz.py", line 110, in do_POST
    text = self.can_engine.call_module(self.can_engine.find_module(str(path_parts[3])), str(paramz))
  File "/home/pop/can/CANToolz/cantoolz/engine.py", line 97, in call_module
    ret = self._enabledList[x][1].raw_write(params)
  File "/home/pop/can/CANToolz/cantoolz/module.py", line 58, in raw_write
    ret = cmd[3](cmd[5], lost_cmd)
  File "/home/pop/can/CANToolz/cantoolz/modules/hw_CANSocket.py", line 40, in dev_write
    dataf = bytes.fromhex(dataf)
ValueError: non-hexadecimal number found in fromhex() arg at position 0
```

**Comment by eik00d** [comment]
will work on this issue this week. No promises, but latest - next week will be fixed. Thank you one more time!

**Comment by eik00d** [comment]
Fixed at #88
